candidatetype.go

```diff
@@ -38,7 +38,21 @@ func (c CandidateType) String() string {
 // The RECOMMENDED values are 126 for host candidates, 100
 // for server reflexive candidates, 110 for peer reflexive candidates,
 // and 0 for relayed candidates.
-func (c CandidateType) Preference() uint16 {
+func (c CandidateType) Preference(networkType NetworkType) uint16 {
```

Comment:
I dont think this exactly reflects the recommendation of [RFC6544 Section 4.2](https://www.rfc-editor.org/rfc/rfc6544.html#section-4.2):

> When both UDP and TCP candidates are offered for the same media stream, and one transport protocol should be preferred over the other, the type preferences for the preferred transport protocol candidates SHOULD be increased and/or the type preferences for the other transport protocol candidates SHOULD be decreased.

So I think we should have a user-configurable offset which is added/substracted from the type-dependent preference.

> How much the values should be increased or decreased depends on whether it is more important to choose a certain transport protocol or a certain candidate type. If the candidate type is more important (e.g., even if UDP is preferred, TCP host candidates are preferred over UDP server reflexive candidates) changing type preference values by one for the other transport protocol candidates is enough.  On the other hand, if the transport protocol is more important (e.g., any UDP candidate is preferred over any TCP candidate), all the preferred transport protocol candidates SHOULD have type preference higher than the other transport protocol candidates.  However, it is RECOMMENDED that the relayed candidates are still preferred lower than the other candidate types. For RTP-based media streams, it is RECOMMENDED that UDP candidates are preferred over TCP candidates.

Reply:
That would mean we need to change existing default type priority for UDP.
Because currently host UDP type priority is 126, max value. So we can't increase it anymore.
If we want to use offsets, we need to make lower.

That may affect existing applications. Personally I would not want to affect existing library clients if they don't use TCP at all.

Reply:
See also https://github.com/pion/ice/commit/92be181f41dde956502806bc283aba289dccca7a

Reply:
Though we may use offset to reduce TCP priorities and not change UDP ones. That would conform to the standard.